crates/router/src/connector/checkout/transformers.rs

```diff
@@ -140,6 +140,34 @@ pub enum CheckoutPaymentStatus {
     Captured,
 }
 
+impl From<transformers::Foreign<(CheckoutPaymentStatus, Option<enums::CaptureMethod>)>>
+    for transformers::Foreign<enums::AttemptStatus>
+{
+    fn from(
+        item: transformers::Foreign<(CheckoutPaymentStatus, Option<enums::CaptureMethod>)>,
+    ) -> Self {
+        let item = item.0;
+        let status = item.0;
+        let capture_method = item.1;
+        match status {
+            CheckoutPaymentStatus::Authorized => {
+                if capture_method == Some(enums::CaptureMethod::Automatic)
+                    || capture_method.is_none()
```

Comment:
This capture method we are getting from request right ?

When doing a sync call, in request capture_method is `None` . And it would mapped to `Charged`, but it should be mapped to `Authorized`. 

Getting the capture method from DB , through router data would avoid that issue.

Reply:
For sync call we are not using this From implementation, We would be using the one with balances.

Reply:
But if automatic capture payment call has happened and then immediately a sync call. 

Then in the response balances[available_to_capture] will be non zero and the status is returned as Authorized. 
Unless we have a guard saying if payment_intent status is `succeeded, don't call connector sync api`. I don't think we have this guard since some succeeded can become failure cc. @jarnura .

Can you check how it behaves for this case. If it is changing status from succeeded to requires_capture . Then I think we can store and get capture method info from DB(attempt table).

Reply:
It shows succeeded @Abhicodes-crypto

Reply:
After status reaches terminal status we won't sync.